Documentation/gettingstarted/clustermesh/aks-clustermesh-prep.rst

```diff
@@ -0,0 +1,235 @@
+AKS(Azure Kubernetes Service) clusters in BYOCNI mode to meet the requirements 
+for the clustermesh feature.
+
+In this guide we will install two AKS clusters in BYOCNI(Bring Your Own CNI) 
+mode and connect them together via clustermesh. This guide is not 
+applicable for cross-cloud clustermesh since 
+
+.. note::
+
+        BYOCNI requires the ``aks-preview`` CLI extension with version >=
+        0.5.55, which itself requires an ``az`` CLI version >= 2.32.0 .
+
+Install cluster one
+###################
+
+1. create a resource group for the cluster(or set the environment variables
+to an existing resource group).
+
+    .. code-block:: bash
+
+        export NAME="$(whoami)-$RANDOM"
+        export AZURE_RESOURCE_GROUP="${NAME}-group"
+
+        # westus2 can be changed to location available (`az account list-locations`)
+        az group create --name "${AZURE_RESOURCE_GROUP}" -l westus2
+
+2. Now that we have a resource group we can create a VNet(virtual network). 
+Creating a custom VNet is required so we can specify a unique Node, Pod, and 
+Service CIDRs to make sure we don't overlap with other clusters.
+
+    .. note::
+        In this case we use the ``192.168.10.0/24`` range, but this can be exchanged
+        for any range except for(``169.254.0.0/16``, ``172.30.0.0/16``, 
+        ``172.31.0.0/16``, or ``192.0.2.0/24`` which are reserved by Azure).
+
+    .. code-block:: bash
+
+        az network vnet create \
+            --resource-group "${AZURE_RESOURCE_GROUP}" \
+            --name "${NAME}-cluster-net" \
+            --address-prefixes 192.168.10.0/24 \
+            --subnet-name "${NAME}-node-subnet" \
+            --subnet-prefix 192.168.10.0/24
+
+        # Store the ID of the created subnet
+        export NODE_SUBNET_ID=$(az network vnet subnet show \
+            --resource-group \"${AZURE_RESOURCE_GROUP}" \
+            --vnet-name "${NAME}-cluster-net" \
+            --name "${NAME}-node-subnet" \
+            --query id \
+            -o tsv)
+
+3. We now have a virtual network and a subnet with the same CIDR. We can now 
+create a AKS cluster without CNI and request to use our custom VNet and subnet.
+
+    During creation we also request to use ``"10.10.0.0/16`` as the pod CIDR and
+    ``"10.11.0.0/16`` as the services CIDR. These can be changed to any range
+    except for Azure reserved ranges and ranges used by other clusters we intend to
+    add to the clustermesh.
+
+    .. code-block:: bash
+
+        az aks create \
+            --resource-group "${AZURE_RESOURCE_GROUP}" \
+            --name "${NAME}" \
+            --network-plugin none \
+            --pod-cidr "10.10.0.0/16" \
+            --service-cidr "10.11.0.0/16" \
+            --dns-service-ip "10.11.0.10" \
+            --vnet-subnet-id "${NODE_SUBNET_ID}"
+
+        # Get kubectl credentials and add them to ~/.kube/config
+        az aks get-credentials \
+            --resource-group "${AZURE_RESOURCE_GROUP}" \
+            --name "${NAME}"
```

Comment:
Maybe provide directly the command? If easy to do.

```suggestion
        AKS_CREDENTIALS=$(az aks get-credentials \
            --resource-group "${AZURE_RESOURCE_GROUP}" \
            --name "${NAME}")
        echo ... something something ${AKS_CREDENTIALS} ... >> "${HOME}/.kube/config"
```

Reply:
The command already does this, it automatically merges the new creds into the config file. I updated the comment above to make this more clear